colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -26,6 +29,30 @@ object HttpMethod {
 
   val methods: List[HttpMethod] = List(Get, Post, Put, Delete, Head, Options, Trace, Connect, Patch)
 
+  def apply(line: Array[Byte]): HttpMethod = {
```

Comment:
There are now tests.  Plus I managed to rework the parsing so it actually really checks the method instead of just counting how many bytes it is.  Turns out `Array.size` is _way_ slower than `Array.length`, since `.size` brings in all the abstract wrappers and such.